ls1intum/Artemis
# `Development`: Add a Java migration tool

### Checklist
#### General


#### Server

### Motivation and Context


During development, we decide on certain changes that might require consistent transformations in the data of the production environment. Liquibase can't do that for us as it focuses on structural changes.
This PR introduces a new tool addressing this issue.

### Description

I created a Registry for different common change entries that get executed. I added an integrity check for the changelog to prevent accidental changes and added tests and documentation.
I additionally added repository, entity and Liquibase changelog to save executed changes as we don't want to execute them multiple times.
Since I was already doing the documentation for my tool, I also added a short section for Liquibase.
I increased the maximal missed class count because the abstract class for the entries is not tested as it doesn't contain any funcitonality.

### Steps for Testing

Please test this locally.
Prerequisites:
- Local Setup

1. Check the documentation
2. Create some Change entries. You can do actual changes or just print things and check in the console
3. Start the server and check the logs for issues
4. Try to "accidentally" break the changelog and make sure that the changelog doesn't run through
5. Delete some entries and check that they are executed again after a restart
6. Do anything else that seems reasonable to you.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
